src/excmds.ts

```diff
@@ -3089,48 +3088,63 @@ export function yank(...content: string[]) {
 }
 
 /**
- * Copies a string to the clipboard/selection buffer depending on the user's preferences
+ * Copies a string to the clipboard/selection buffer depending on the user's preferences.
  *
  * @hidden
  */
 //#background_helper
-async function setclip(str) {
-    // Functions to avoid retyping everything everywhere
+async function setclip(data: string) {
+    // Function to avoid retyping everything everywhere
+    const setclip_selection = () => Native.clipboard("set", data)
 
-    // Note: We're using fillcmdline here because exceptions are somehow not caught. We're rethrowing because otherwise the error message will be overwritten with the "yank successful" message.
-    const s = () => Native.clipboard("set", str)
-    const c = () => messageActiveTab("commandline_frame", "setClipboard", [str])
-
-    let promises = []
+    let promises: Array<Promise<any>> = []
     switch (await config.getAsync("yankto")) {
         case "selection":
-            promises = [s()]
+            promises = [setclip_selection()]
             break
         case "clipboard":
-            promises = [c()]
+            promises = [setclip_api(data)]
             break
         case "both":
-            promises = [s(), c()]
+            promises = [setclip_selection(), setclip_api(data)]
             break
     }
     return Promise.all(promises)
 }
 
+/**
+ * Copies a string to the clipboard using the Clipboard API.
+ * @hidden
+ */
+//#background_helper
+async function setclip_api(data: string) {
```

Comment:
```suggestion
async function setclip_webapi(data: string) {
```

The _api confused me as the primary selection also uses an API (arguably moreso since it's crossing a process boundary) : )